src/cpp/r/RCntxt.cpp

```diff
@@ -111,12 +111,24 @@ SEXP RCntxt::originalFunctionCall() const
 
 Error RCntxt::fileName(std::string* pFileName) const
 {
-   if (srcref() && TYPEOF(srcref()) != NILSXP)
+   // skip over bytecode srcrefs
+   RCntxt context = *this;
+   SEXP ref = context.srcref();
+   while (ref && isByteCodeSrcRef(ref))
```

Comment:
Okay, let's take as-is then. As this will need to get cherry-picked for the patch release we will probably want to have Darby do some focused testing on the debugger to make sure the refs line up properly.